actions.py

```diff
@@ -245,13 +245,20 @@ def validate_time(
 
         def get_duration(entity):
             value = entity.get("value")
-            start = entity.get("from")
-            end = entity.get("to")
-            grain = entity.get("additional_info").get("grain")
-
+            try:
+                start = value.get("from")
+                end = value.get("to")
+                grain = entity.get("additional_info").get("from").get("grain")
+                reportgrain = "timeframe"
```

Comment:
why do we not use the grain here?

Reply:
When the user specifies a `from` and `to` value, the grain usually doesn't make sense to return in the message e.g. `I want to search my transaction history from jan 1 till january 14` has `grain` of `day`, but it doesn't make sense to say `Searching transactions during the day of Friday Jan 01, 2021 to Friday Jan 15, 2021...`. That's why there is a `reportgrain` for the response message and a `grain` which is used for finding an `end` (if it wasn't specified).

Reply:
ok makes sense - but I think `start` and `end` should go outside of this try/except right?

Reply:
`value` will be a string if it is not a dict leading to `AttributeError`, hence why it is all inside the try/except.

Reply:
whoops, sorry i missed that